Deprecated: DexLib 2.0.3

Hey @david-allison-1 - out of curiosity it appears we are on 2.0.3 of dexlib from 2014 https://github.com/JesusFreke/smali/releases/tag/v2.0.3

And when dependabot attempts to move it to current, it fails

```
Task 
:AnkiDroid:compileDebugAndroidTestJavaWithJavac FAILED

/home/travis/build/mikehardy/Anki-Android/AnkiDroid/src/androidTest/java/androidx/test/internal/runner/MultiDexClassPathScanner.java:82: error: no suitable method found for loadDexFile(String,int)

            DexBackedDexFile dex = DexFileFactory.loadDexFile(dexPath, Build.VERSION.SDK_INT);

                                                 ^

    method DexFileFactory.loadDexFile(String,Opcodes) is not applicable

      (argument mismatch; int cannot be converted to Opcodes)

    method DexFileFactory.loadDexFile(File,Opcodes) is not applicable

      (argument mismatch; String cannot be converted to File)

1 error
```

Curious if the version choice was intentional / upgrade is impossible or if an upgrade is possible?

https://github.com/mikehardy/Anki-Android/pull/211

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -180,5 +180,5 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-contrib:3.2.0'
     androidTestImplementation 'androidx.test.ext:junit:1.1.1'
     androidTestImplementation 'androidx.test:rules:1.2.0'
-    androidTestImplementation 'org.smali:dexlib2:2.0.3'
+    androidTestImplementation 'org.smali:dexlib2:2.4.0'
 }

--- a/AnkiDroid/src/androidTest/java/androidx/test/internal/runner/MultiDexClassPathScanner.java
+++ b/AnkiDroid/src/androidTest/java/androidx/test/internal/runner/MultiDexClassPathScanner.java
@@ -22,6 +22,7 @@ import android.os.Build;
 import com.ichi2.libanki.Utils;
 
 import org.jf.dexlib2.DexFileFactory;
+import org.jf.dexlib2.Opcodes;
 import org.jf.dexlib2.dexbacked.DexBackedDexFile;
 import org.jf.dexlib2.iface.ClassDef;
 
@@ -79,7 +80,7 @@ class MultiDexClassPathScanner extends ClassPathScanner {
     private List<String> extractClassesFromDexPath(String dexPath) {
         List<String> ret = new ArrayList<>();
         try {
-            DexBackedDexFile dex = DexFileFactory.loadDexFile(dexPath, Build.VERSION.SDK_INT);
+            DexBackedDexFile dex = DexFileFactory.loadDexFile(new File(dexPath), Opcodes.forApi(Build.VERSION.SDK_INT));
             for (ClassDef classDef: dex.getClasses()) {
                 String typeName = extractTypeNameFromDef(classDef);
                 ret.add(typeName);

